public/app/features/dashboard-scene/settings/JsonModelEditView.tsx

```diff
@@ -31,11 +43,27 @@ export class JsonModelEditView extends SceneObjectBase<JsonModelEditViewState> i
     return getPrettyJSON(jsonData);
   }
 
+  public onCodeEditorBlur = (value: string) => {
+    this.setState({ jsonText: value });
+  };
+
+  public onApplyChange = () => {
+    const jsonModel = JSON.parse(this.state.jsonText);
+    const dashboard = this.getDashboard();
+    const rsp: DashboardDTO = {
+      dashboard: jsonModel,
+      meta: dashboard.state.meta,
+    };
+    const newDashboardScene = transformSaveModelToScene(rsp);
+    const newState = sceneUtils.cloneSceneObjectState(newDashboardScene.state);
+    dashboard.setState(newState);
```

Comment:
doing this used to cause a warning that a child already belongs to a dashboard scene on saving the dashboard, however the warning somehow doesn't show up anymore. i'm curious to know if it will show up for others. when we save this method is triggered:

https://github.com/grafana/grafana/blob/9d8f4228bb63355f065637007e1d7e03227362f6/public/app/features/dashboard-scene/scene/DashboardScene.tsx#L205-L207

and I'm wondering if it was pulling ref from the old dashboard before we passed new state to it

Reply:
I am not seeing any warning. I have seen that a few times I cannot exit dashboard settings (via breadcrumb) after hitting "Save changes"